Source/SessionManager/SessionManager+Backup.swift

```diff
@@ -41,17 +52,34 @@ extension SessionManager {
         case unknown
     }
 
-    public func backupActiveAccount(password: String, completion: @escaping BackupResultClosure) {
+    public func backupActiveAccount(password: String, completion: @escaping BackupResultClosure) throws {
         guard let userId = accountManager.selectedAccount?.userIdentifier,
               let clientId = activeUserSession?.selfUserClient?.remoteIdentifier,
               let handle = activeUserSession.flatMap(ZMUser.selfUser)?.handle else { return completion(.failure(BackupError.noActiveAccount)) }
+        
+        var encryptionKeys: EncryptionKeys?
+        if activeUserSession?.storeProvider.contextDirectory.uiContext.encryptMessagesAtRest ?? false {
+            do {
+                encryptionKeys = try activeUserSession?.storeProvider.contextDirectory.uiContext.getEncryptionKeys()
+            } catch {
+                throw EARBackupError.missingEAREncryptionKey
+            }
+        }
 
         StorageStack.backupLocalStorage(
             accountIdentifier: userId,
             clientIdentifier: clientId,
             applicationContainer: sharedContainerURL,
             dispatchGroup: dispatchGroup,
-            completion: { [dispatchGroup] in SessionManager.handle(result: $0, password: password, accountId: userId, dispatchGroup: dispatchGroup, completion: completion, handle: handle) }
+            encryptionKeys: encryptionKeys,
```

Comment:
```suggestion
            encryptionKeys: activeUserSession.managedObjectContext.encryptionKeys,
```

Could also access the `encryptionKeys` like this and then you don't need deal with the error here.